src/base/default_coupling.C

```diff
@@ -78,7 +78,12 @@ void DefaultCoupling::operator()
 {
   LOG_SCOPE("operator()", "DefaultCoupling");
 
-  libmesh_assert(_mesh);
+  // Let us not do assertion at this moment for API upgrade.
+  // There is a functor inside of ElementSideNeighborLayers.
+  // We can not set mesh for that functor because there is no handle
+  // in libmesh. We need to override set_mesh in moose for setting a mesh for the functor.
```

Comment:
Ah - set_mesh() is getting called by MOOSE *only* if _app.getExecutioner() returns non-null?

So we can leave the check that comes later in the get_periodic_bcs case, because that's actually checking if something's broken, but the check here would be overzealous?  Makes sense to me.

Reply:
So we will be able to uncomment this eventually though right?

Reply:
Yes, we will uncomment all these.